### PR TITLE
Fix cmsearch

### PR DESCRIPTION
**Pull request by KateSakharova**
Add chunking to cmsearch sub-wf for WGS and amplicon

concatenate models -> chunking fasta on 2mnl seqs -> scatter cmsearch -> scatter deoverlap -> cat cmsearch result [output] --> cat deoverlap result [output]

### CODE REVIEW

File: utils/concatenate.cwl [hunk 1]
```diff
@@ -24,7 +26,11 @@ stdout: $(inputs.outputFileName)$(inputs.postfix)
 outputs:
   - id: result
     type: stdout
-    format: ${if ("format" in inputs.files[0]) return inputs.files[0].format; else return 'undefined'}
+    format: ${if (typeof(inputs.files[0]) == 'string') return 'underfined';
```
Comment (mberacochea) [inline utils/concatenate.cwl]: Typo `underfined`. Also shouldn't that be `undefined` with no quotes?

File: workflows/subworkflows/cmsearch-condition.cwl [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+class: Workflow
+$namespaces:
+  edam: 'http://edamontology.org/'
+  s: 'http://schema.org/'
+label: Identifies non-coding RNAs using Rfams covariance models
+
+requirements:
+  SubworkflowFeatureRequirement: {}
+  MultipleInputFeatureRequirement: {}
+  InlineJavascriptRequirement: {}
```
Comment (mberacochea) [inline workflows/subworkflows/cmsearch-condition.cwl]: I think you don't need InlineJavascriptRequirement nor ScatterFeatureRequirement?

File: workflows/subworkflows/cmsearch-multimodel-raw-data.cwl [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+        source: query_sequences
+        valueFrom: $(self.nameroot)
+      postfix: { default: ".cmsearch.all.tblout.deoverlapped" }
+    out: [ result ]
+
+$schemas:
+  - 'http://edamontology.org/EDAM_1.16.owl'
+  - 'https://schema.org/version/latest/schemaorg-current-http.rdf'
+s:license: "https://www.apache.org/licenses/LICENSE-2.0"
+s:copyrightHolder: "EMBL - European Bioinformatics Institute, 2018"
```
Comment (mberacochea) [inline workflows/subworkflows/cmsearch-multimodel-raw-data.cwl]: 2020?

File: tools/RNA_prediction/cmsearch/infernal-cmsearch-v1.1.2.cwl [hunk 4]
```diff
@@ -64,13 +64,29 @@ outputs:
     type: File
     format: edam:format_3475
     outputBinding:
-      glob: $(inputs.query_sequences.basename).$(inputs.covariance_model_database.split('/').slice(-1)[0]).cmsearch_matches.tbl
+      glob: |
+        ${
```
Comment (mberacochea) [inline tools/RNA_prediction/cmsearch/infernal-cmsearch-v1.1.2.cwl:68]: May I suggest we format this code to improve readability:
```js
var name = "";
if (typeof inputs.covariance_model_database === "string") {
  name =
    inputs.query_sequences.basename +
    "." +
    inputs.covariance_model_database.split("/").slice(-1)[0] +
    ".cmsearch_matches.tbl";
} else {
  name =
    inputs.query_sequences.basename +
    "." +
    inputs.covariance_model_database.nameroot +
    ".cmsearch_matches.tbl";
}
return name;
```

File: tools/RNA_prediction/cmsearch/infernal-cmsearch-v1.1.2.cwl [hunk 5]
```diff
@@ -64,13 +64,29 @@ outputs:
...
+          return name;
+        }
   - id: programOutput
     label: 'direct output to file, not stdout'
     type: File
     format: edam:format_3475
     outputBinding:
-      glob: $(inputs.query_sequences.basename).$(inputs.covariance_model_database.split('/').slice(-1)[0]).cmsearch.out
+      glob: |
+        ${
```
Comment (mberacochea) [inline tools/RNA_prediction/cmsearch/infernal-cmsearch-v1.1.2.cwl:91]:
```js
var name = "";
if (typeof inputs.covariance_model_database == "string") {
  name =
    inputs.query_sequences.basename +
    "." +
    inputs.covariance_model_database.split("/").slice(-1)[0] +
    ".cmsearch.out";
} else {
  name =
    inputs.query_sequences.basename +
    "." +
    inputs.covariance_model_database.nameroot +
    ".cmsearch.out";
}
return name;
```

File: tools/RNA_prediction/cmsearch/infernal-cmsearch-v1.1.2.cwl [hunk 6]
```diff
@@ -90,10 +106,32 @@ label: Search sequence(s) against a covariance model database
 arguments:
   - position: 0
     prefix: '--tblout'
-    valueFrom: $(inputs.query_sequences.basename).$(inputs.covariance_model_database.split('/').slice(-1)[0]).cmsearch_matches.tbl
+    valueFrom: |
+      ${
+        var name = "";
```
Comment (mberacochea) [inline tools/RNA_prediction/cmsearch/infernal-cmsearch-v1.1.2.cwl:129]:
```js
var name = "";
if (typeof inputs.covariance_model_database == "string") {
  name =
    inputs.query_sequences.basename +
    "." +
    inputs.covariance_model_database.split("/").slice(-1)[0] +
    ".cmsearch_matches.tbl";
} else {
  name =
    inputs.query_sequences.basename +
    "." +
    inputs.covariance_model_database.nameroot +
    ".cmsearch_matches.tbl";
}
return name;
```

File: tools/RNA_prediction/cmsearch/infernal-cmsearch-v1.1.2.cwl [hunk 7]
```diff
@@ -90,10 +106,32 @@ label: Search sequence(s) against a covariance model database
...
+          name = inputs.query_sequences.basename + '.' + inputs.covariance_model_database.split('/').slice(-1)[0] + '.cmsearch_matches.tbl'; }
+        else {
+          name = inputs.query_sequences.basename + '.' + inputs.covariance_model_database.nameroot + '.cmsearch_matches.tbl'; }
+        return name;
+      }
   - position: 0
     prefix: '-o'
-    valueFrom: $(inputs.query_sequences.basename).$(inputs.covariance_model_database.split('/').slice(-1)[0]).cmsearch.out
+    valueFrom: |
+      ${
```
Comment (mberacochea) [inline tools/RNA_prediction/cmsearch/infernal-cmsearch-v1.1.2.cwl:148]:
```js
var name = "";
if (typeof inputs.covariance_model_database == "string") {
  name =
    inputs.query_sequences.basename +
    "." +
    inputs.covariance_model_database.split("/").slice(-1)[0] +
    ".cmsearch.out";
} else {
  name =
    inputs.query_sequences.basename +
    "." +
    inputs.covariance_model_database.nameroot +
    ".cmsearch.out";
}
return name;
```

File: utils/concatenate.cwl [hunk 8]
```diff
@@ -24,7 +26,11 @@ stdout: $(inputs.outputFileName)$(inputs.postfix)
...
+            else {
+                   if ("format" in inputs.files[0]) return inputs.files[0].format;
+                   else return 'underfined'}
```
Comment (mberacochea) [inline utils/concatenate.cwl]: Typo underfined. Also shouldn't that be undefined with no quotes?

File: utils/concatenate.cwl (hunk 1 above)
Comment (KateSakharova) [reply utils/concatenate.cwl]: It seems working with quotes ...

**Pull request closed without merging**